### PR TITLE
Fix logcollector skipped tests

### LINKED ISSUE #2286: Fix logcollector skipped tests
| Development branch | 
|--|
`2286-fix-logcollector-skips`

### Description

After https://github.com/wazuh/wazuh-qa/pull/2185 we have achieved to sanitized most of the logcollector tests. In order to achieve full-green in the master branch, [we had skipped unstable tests](https://github.com/wazuh/wazuh-qa/pull/2185#issuecomment-981408814). 

We need to investigate these false positives in order to achieve a real full-green in the logcollector module.

#### List of skipped tests
- [x] test_command_execution_dbg @danisan90, @Rebits 
- [x] test_location_custom_sockets @fernandolojano, @Rebits 
- [x] test_macos_file_status_predicate @Rebits 
- [x] test_macos_log_process @Rebits 
- [x] test_options_state_interval_no_file @Rebits 

### Testing
A full report of logcollector for Manager/Agent in every supported OS (Solaris, Centos8, macOS, Windows) is necessary in order to close this issue.

#### Package
| Version | Revision | Link|
|---|---|---|
|4.3|0.40400.20211129| https://ci.wazuh.info/job/Packages_builder_tier/1828/console

### PR DESCRIPTION
**Pull request by Rebits**
|Related issue|
|---|
|#2286|

### Description

This PR closes #2286 

| Version | Revision | Link|
|---|---|---|
|4.3|0.40400.20211129| https://ci.wazuh.info/job/Packages_builder_tier/1828/console

### Logcollector

|  CentOS: Manager  | Local   | Jenkins     | Notes
|---    |---   |---    |--    |
| PS1  | [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/7683354/PS1-2301-centos-manager.tar.gz)  | [:green_circle: ](https://ci.wazuh.info/job/Test_integration/17108/)  |
| PS2   | [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/7683355/PS2-2301-centos-manager.tar.gz)  | [:green_circle: ](https://ci.wazuh.info/job/Test_integration/17109/console) | 
| PS3   |  [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/7683356/PS3-2301-centos-manager.tar.gz)  | [:green_circle: ](https://ci.wazuh.info/job/Test_integration/17110/console) |

* * * 

|  CentOS: Agent  | Local   | Jenkins     | Notes
|---    |---   |---    |--    |
| PS1  | [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/7691642/PS1-2301-centos-agent.tar.gz) | [:green_circle: ](https://ci.wazuh.info/job/Test_integration/17113/console)
| PS2   | [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/7691643/PS2-2301-centos-agent.tar.gz) | [:green_circle: ](https://ci.wazuh.info/job/Test_integration/17114/console) | 
| PS3   |  [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/7691644/PS3-2301-centos-agent.tar.gz)  | [:green_circle: ](https://ci.wazuh.info/job/Test_integration/17115/console)|

* * * 

|  macOS: agent  | Local   | Jenkins     | Notes
|---    |---   |---    |--    |
| PS1  | [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/7691688/R1-macos-agent-2301.zip)  |  :orange_circle:   |
| PS2   |  [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/7692085/R2-macos-agent-logcollector.zip)  | :orange_circle:  | 
| PS3   |  [:green_circle:](https://github.com/wazuh/wazuh-qa/files/7693967/R3-macos-agent.zip) | :orange_circle:  |

* * *

|  Windows: Agent | Local   | Jenkins     | Notes
|---    |---   |---    |--    |
| PS1  | [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/7683334/PS3-2286-win-agent.zip) |  [:green_circle:  ](https://ci.wazuh.info/job/Test_integration/17317/console)
| PS2   | [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/7683336/PS2-2286-win-agent.zip) | [:green_circle:  ](https://ci.wazuh.info/job/Test_integration/17316/console) | 
| PS3   | [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/7683337/PS1-2286-win-agent.zip) | [:green_circle:  ](https://ci.wazuh.info/job/Test_integration/17315/console) |

* * *

|  Solaris: Agent  | Local   | Jenkins     | Notes
|---    |---   |---    |--    |
| PS1  | [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/7684980/PS1-2301-solaris-agent.zip)  | :orange_circle:   |
| PS2   |  [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/7684981/PS2-2301-solaris-agent.zip) | :orange_circle:  | 
| PS3   |  [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/7684982/PS3-2301-solaris-agent.zip)  | :orange_circle:  |

* * * 

* :green_circle:: All pass
* :yellow_circle:: Some warnings
* :red_circle:: Some errors/fails
* :large_blue_circle:: In progress 
* :orange_circle: Jenkins provision fails.

### CODE REVIEW

**Comment by Rebits** [comment]
#10/12/21

Due to the [current bug in macOS/Solaris Jenkins deployment](https://github.com/wazuh/wazuh-jenkins/issues/3115), the logcollector module is going to be tested only in local for that operative system.

File: tests/integration/test_logcollector/test_location/test_location.py [hunk 1]
```diff
@@ -105,8 +105,8 @@
      'files': [os.path.join(temp_dir, 'wazuh-testing', 'duplicated', 'duplicated.txt')],
      'log_format': 'syslog', 'path_2': os.path.join(temp_dir, 'wazuh-testing', 'duplicated', 'duplicated.txt'),
      'file_type': 'duplicated_file'},
-    {'location': os.path.join(temp_dir, 'wazuh-testing', 'file.log-%Y-%m-%d'),
-     'files': [os.path.join(temp_dir, 'wazuh-testing',f'file.log-{date}')], 'log_format': 'syslog',
+    {'location': os.path.join(temp_dir, 'wazuh-testing', r'file.log-%Y-%m-%d'),
+     'files': [os.path.join(temp_dir, 'wazuh-testing', f'file.log-{date}')], 'log_format': 'syslog',
```
Comment (juliamagan) [inline tests/integration/test_logcollector/test_location/test_location.py:109]:
```suggestion
     'files': [os.path.join(temp_dir, 'wazuh-testing', f"file.log-{date}")], 'log_format': 'syslog',

```

File: tests/integration/test_logcollector/test_macos/data/wazuh_macos_format_only_future_events.yaml [hunk 2]
```diff
@@ -11,3 +11,5 @@
           value: 'macos'
       - only-future-events:
           value: ONLY_FUTURE_EVENTS
+      - query:
+          value: 'process = "logger"'
```
Comment (juliamagan) [inline tests/integration/test_logcollector/test_macos/data/wazuh_macos_format_only_future_events.yaml:15]:
```suggestion
          value: 'process = "logger"'
          
```

File: tests/integration/test_logcollector/test_macos/test_macos_log_process.py [hunk 3]
```diff
@@ -142,7 +143,7 @@ def test_macos_log_process_stop_suddenly_warning(get_configuration, configure_en
 
     macos_logcollector_monitored = logcollector.callback_monitoring_macos_logs
     log_monitor.start(timeout=logcollector.LOG_COLLECTOR_GLOBAL_TIMEOUT, callback=macos_logcollector_monitored,
-                            error_message=logcollector.GENERIC_CALLBACK_ERROR_TARGET_SOCKET)
+                      error_message=logcollector.GENERIC_CALLBACK_ERROR_TARGET_SOCKET)
     time.sleep(5)
```
Comment (juliamagan) [inline tests/integration/test_logcollector/test_macos/test_macos_log_process.py:147]: Hardcoded sleeps are not allowed, we should create a variable with a descriptive name.

File: tests/integration/test_logcollector/test_options/test_options_state_interval_no_file.py [hunk 4]
```diff
@@ -1,154 +1,168 @@
...
+def get_local_internal_options_function(request):
+    """Get configurations from the module."""
+    global local_internal_options
+    local_internal_options = {'logcollector.open_attempts': request.param['open_attempts'],
+                              'logcollector.state_interval': request.param['state_interval'],
+                              'logcollector.vcheck_files': '3',
+                              'logcollector.debug': '2'}
+
+
+def test_options_state_interval_no_file(get_local_internal_options_function, configure_local_internal_options_function,
```
Comment (juliamagan) [inline tests/integration/test_logcollector/test_options/test_options_state_interval_no_file.py:96]: Review hardcoded timeouts, please.

File: tests/integration/test_logcollector/test_options/test_options_state_interval_no_file.py [hunk 5]
```diff
@@ -1,154 +1,168 @@
...
+            sleep(time_to_update_statistics_file)
+
+            if use_regex:
+                assert not any(check_wazuh_logcollector_status_file(log_path)), f"Using regex as location, file \
+                                                                                 {location} has not been deleted from \
+                                                                                 {LOGCOLLECTOR_STATISTICS_FILE}"
+            else:
+                assert all(check_wazuh_logcollector_status_file(log_path)),  f"Using hardcoded location, file \
+                                                                               {log_path} has been deleted \
+                                                                               sfrom {LOGCOLLECTOR_STATISTICS_FILE}"
```
Comment (juliamagan) [inline tests/integration/test_logcollector/test_options/test_options_state_interval_no_file.py:168]:
```suggestion
                assert all(check_wazuh_logcollector_status_file(log_path)),  f"Using hardcoded location, file \
                                                                               {log_path} has been deleted \
                                                                               from {LOGCOLLECTOR_STATISTICS_FILE}"
```

File: tests/integration/test_logcollector/test_command_monitoring/test_command_execution_dbg.py [hunk 6]
```diff
@@ -118,12 +120,12 @@ def dbg_reading_command(command, alias, log_format):
...
         msg = fr"DEBUG: Reading command message: 'ossec: output: '{alias}': {output}'"
 
-    wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
+    log_monitor.start(timeout=global_parameters.default_timeout,
                       callback=monitoring.make_callback(pattern=msg, prefix=prefix),
                       error_message=logcollector.GENERIC_CALLBACK_ERROR_COMMAND_MONITORING)
 
 
-@pytest.mark.skip(reason="Unexpected false positive, further investigation is required")
+@pytest.mark.xfail(reason='Expected error: https://github.com/wazuh/wazuh/issues/11192')
```
[the remark this answers is not in the document]
Comment (Rebits) [reply tests/integration/test_logcollector/test_command_monitoring/test_command_execution_dbg.py]: Done 9c61d46.New issue/pr has been created for this case.

File: tests/integration/test_logcollector/test_configuration/test_basic_configuration_exclude.py [hunk 7]
```diff
@@ -5,7 +5,9 @@
 import os
 import sys
 import pytest
+from time import sleep
 import wazuh_testing.api as api
+
 from wazuh_testing.tools.services import get_service
 from wazuh_testing.tools.configuration import load_wazuh_configurations
 from wazuh_testing.tools.services import get_process_cmd, check_if_process_is_running
```
[the remark this answers is not in the document]
Comment (Rebits) [reply tests/integration/test_logcollector/test_configuration/test_basic_configuration_exclude.py]: Done 9c61d46, general imports had been separated from `wazuh_testing` ones.

File: tests/integration/test_logcollector/test_location/test_location.py [hunk 8]
```diff
@@ -23,7 +23,7 @@
 local_internal_options = {'logcollector.debug': '2'}
 
 temp_dir = tempfile.gettempdir()
-date = datetime.date.today().strftime("%Y-%m-%d")
+date = datetime.date.today().strftime(r"%Y-%m-%d")
```
[the remark this answers is not in the document]
Comment (Rebits) [reply tests/integration/test_logcollector/test_location/test_location.py]: Done 9c61d46

File: tests/integration/test_logcollector/test_location/test_location.py (hunk 1 above)
Comment (Rebits) [reply tests/integration/test_logcollector/test_location/test_location.py]: Done 9c61d46

File: tests/integration/test_logcollector/test_macos/data/wazuh_macos_format_only_future_events.yaml (hunk 2 above)
Comment (Rebits) [reply tests/integration/test_logcollector/test_macos/data/wazuh_macos_format_only_future_events.yaml]: Done 9c61d46

File: tests/integration/test_logcollector/test_macos/test_macos_format_only_future_events.py [hunk 9]
```diff
@@ -78,29 +82,32 @@ def test_macos_format_only_future_events(restart_logcollector_required_daemons_p
...
     expected_old_macos_message = logcollector.format_macos_message_pattern('logger', old_message)
 
+    log_monitor.start(timeout=macos_log_message_timeout,
+                      callback=logcollector.callback_macos_log(expected_old_macos_message))
 
-    log_monitor.start(timeout=macos_log_message_timeout, 
-                              callback=logcollector.callback_macos_log(expected_old_macos_message))
-
-    ## Stop wazuh agent and ensure it gets old macos messages if only-future-events option is disabled
+    # Stop wazuh agent and ensure it gets old macos messages if only-future-events option is disabled
```
[the remark this answers is not in the document]
Comment (Rebits) [reply tests/integration/test_logcollector/test_macos/test_macos_format_only_future_events.py:88]: Done 9c61d46

File: tests/integration/test_logcollector/test_macos/test_macos_log_process.py (hunk 3 above)
Comment (Rebits) [reply tests/integration/test_logcollector/test_macos/test_macos_log_process.py]: Done 9c61d46

File: tests/integration/test_logcollector/test_options/test_options_state_interval_no_file.py (hunk 4 above)
Comment (Rebits) [reply tests/integration/test_logcollector/test_options/test_options_state_interval_no_file.py:96]: Done 9c61d46

File: tests/integration/test_logcollector/test_options/test_options_state_interval_no_file.py (hunk 5 above)
Comment (Rebits) [reply tests/integration/test_logcollector/test_options/test_options_state_interval_no_file.py]: Done 9c61d46

**Comment by juliamagan** [comment]
## 14/12/2021

## Packages

| Version | Revision | Link|
|---|---|---|
|4.3.0|40303| https://packages-dev.wazuh.com/warehouse/test/4.3/rpm/var/wazuh-manager-4.3.0-2286.logcollector.review.x86_64.rpm
|4.3.0|40303| https://packages-dev.wazuh.com/warehouse/test/4.3/rpm/var/wazuh-agent-4.3.0-2286.logcollector.review.x86_64.rpm
|4.3.0|40303| https://packages-dev.wazuh.com/warehouse/test/4.3/solaris/i386/11/wazuh-agent_v4.3.0-2286.logcollector.review-sol11-i386.p5p
|4.3.0|40303| https://packages-dev.wazuh.com/warehouse/test/4.3/windows/wazuh-agent-4.3.0-2286.logcollector.review.msi
|4.3.0|40303| https://packages-dev.wazuh.com/warehouse/test/4.3/macos/wazuh-agent-4.3.0-2286.logcollector.review.pkg

## Testing

### Logcollector

|  CentOS: Manager   | Jenkins     | Notes
|---    |---   |---    |
| R1  | [🔵 ](https://ci.wazuh.info/job/Test_integration/17462/) |  |
| R2   |  |  | 
| R3   |  | |

* * * 

|  CentOS: Agent | Jenkins     | Notes
|---    |---   |---    |
| R1  |  |  |
| R2   |  |  | 
| R3   |  | |

* * * 

|  macOS: Agent | Jenkins     | Notes
|---    |---   |---    |
| R1  |  |  |
| R2   |  |  | 
| R3   |  | |



* * *

|  Windows: Agent   | Jenkins     | Notes
|---    |---   |---    |
| R1  |  |  |
| R2   |  |  | 
| R3   |  | |



* * *

|  Solaris: Agent   | Jenkins     | Notes
|---    |---   |---    |
| R1  |  |  |
| R2   |  |  | 
| R3   |  | |
* * * 

* :green_circle:: All pass
* :yellow_circle:: Some warnings
* :red_circle:: Some errors/fails
* :large_blue_circle:: In progress 
* :orange_circle: Jenkins provision fails.

**Pull request merged**